XR_APILAYER_NOVENDOR_toolkit/layer.cpp

```diff
@@ -1923,6 +1923,30 @@ namespace {
                     correctedProjectionLayer->views = correctedProjectionViews;
                     correctedLayers.push_back(
                         reinterpret_cast<const XrCompositionLayerBaseHeader*>(correctedProjectionLayer));
+                } else if (chainFrameEndInfo.layers[i]->type == XR_TYPE_COMPOSITION_LAYER_QUAD) {
+                    const XrCompositionLayerQuad* quad = reinterpret_cast<const XrCompositionLayerQuad*>(chainFrameEndInfo.layers[i]);
+
+                    auto swapchainIt = m_swapchains.find(quad->subImage.swapchain);
+                    if (swapchainIt == m_swapchains.end()) {
+                        throw std::runtime_error("Swapchain is not registered");
+                    }
+
+                    auto& swapchainState = swapchainIt->second;
+                    auto& swapchainImages = swapchainState.images[swapchainState.acquiredImageIndex];
+
+                    uint32_t nextImage = 0;
+                    uint32_t lastImage = 0;
+
+                    for (size_t i = 0; i < std::size(m_imageProcessors); i++) {
```

Comment:
This could probably just be avoided and do a single copy from 0 (which is always the app texture) to size-1 (always the runtime texture). The intermediate are not needed.